### ISSUE TITLE
Binary model always predict same label

### ISSUE THREAD

**Issue by coccocarmiano** [open-mmlab/mmsegmentation#2579]
I'm trying to implement a binary segmentor based on BiSeNet but during training the model does not seem to learn anything.
Here is a log:

```
2023-02-09 17:52:19,253 - mmseg - INFO - workflow: [('train', 1)], max: 200 iters
2023-02-09 17:52:19,255 - mmseg - INFO - Checkpoints will be saved to /content/mmsegmentation/test_output by HardDiskBackend.
2023-02-09 17:52:31,907 - mmseg - INFO - Saving checkpoint at 10 iterations
2023-02-09 17:52:32,668 - mmseg - INFO - Iter [10/200]	lr: 5.186e-03, eta: 0:04:13, time: 1.337, data_time: 0.815, memory: 6661, decode.loss_ce: 0.4044, decode.acc_seg: 93.6138, aux_0.loss_ce: 0.2625, aux_0.acc_seg: 93.6138, aux_1.loss_ce: 0.2194, aux_1.acc_seg: 93.6138, aux_2.loss_ce: 0.1905, aux_2.acc_seg: 93.6138, aux_3.loss_ce: 0.1516, aux_3.acc_seg: 93.6138, loss: 1.2283

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 18/18, 3.7 task/s, elapsed: 5s, ETA:     0s

2023-02-09 17:52:37,511 - mmseg - INFO - per class results:
2023-02-09 17:52:37,513 - mmseg - INFO - 
+-------+-------+-------+
| Class |  IoU  |  Acc  |
+-------+-------+-------+
|  sane | 97.96 | 100.0 |
|  sick |  0.0  |  0.0  |
+-------+-------+-------+
2023-02-09 17:52:37,517 - mmseg - INFO - Summary:
2023-02-09 17:52:37,519 - mmseg - INFO - 
+-------+-------+------+
|  aAcc |  mIoU | mAcc |
+-------+-------+------+
| 97.95 | 48.98 | 50.0 |
+-------+-------+------+
2023-02-09 17:52:37,521 - mmseg - INFO - Iter(val) [18]	aAcc: 0.9795, mIoU: 0.4898, mAcc: 0.5000, IoU.sane: 0.9796, IoU.sick: 0.0000, Acc.sane: 1.0000, Acc.sick: 0.0000
2023-02-09 17:52:50,433 - mmseg - INFO - Saving checkpoint at 20 iterations
2023-02-09 17:52:51,040 - mmseg - INFO - Iter [20/200]	lr: 5.353e-03, eta: 0:04:45, time: 1.837, data_time: 1.302, memory: 6661, decode.loss_ce: 0.3414, decode.acc_seg: 93.4630, aux_0.loss_ce: 0.2357, aux_0.acc_seg: 93.4630, aux_1.loss_ce: 0.1698, aux_1.acc_seg: 93.4630, aux_2.loss_ce: 0.1163, aux_2.acc_seg: 93.4630, aux_3.loss_ce: 0.1072, aux_3.acc_seg: 93.4630, loss: 0.9704

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 18/18, 4.0 task/s, elapsed: 5s, ETA:     0s

2023-02-09 17:52:55,561 - mmseg - INFO - per class results:
2023-02-09 17:52:55,564 - mmseg - INFO - 
+-------+-------+-------+
| Class |  IoU  |  Acc  |
+-------+-------+-------+
|  sane | 97.96 | 100.0 |
|  sick |  0.0  |  0.0  |
+-------+-------+-------+
2023-02-09 17:52:55,566 - mmseg - INFO - Summary:
2023-02-09 17:52:55,568 - mmseg - INFO - 
+-------+-------+------+
|  aAcc |  mIoU | mAcc |
+-------+-------+------+
| 97.95 | 48.98 | 50.0 |
+-------+-------+------+
```

While this is my cfg file:

```
# All models inherhit from a base in MMSeg
_base_ = "./configs/bisenetv2/bisenetv2_fcn_4x8_1024x1024_160k_cityscapes.py"
num_classes = 1

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True
)
norm_cfg = dict(type="BN", requires_grad=True)  # Single GPU
dataset_type = "MGIDataset"
gpu_ids = [0]
seed = 0
device = "cuda"
model = dict(
    decode_head=dict(
        # SyncBN -> BN
        norm_cfg=norm_cfg,
        # Binary Classification Adjustments
        loss_decode=dict(type="CrossEntropyLoss", use_sigmoid=True, loss_weight=1.0),
        out_channels=num_classes,
        num_classes=num_classes,
    ),
    # This is a copy paste from _base_ with minor adjustments to disable SyncBN
    auxiliary_head=[
        dict(
            type="FCNHead",
            in_channels=16,
            channels=16,
            num_convs=2,
            num_classes=num_classes,
            in_index=1,
            norm_cfg=norm_cfg,
            concat_input=False,
            align_corners=False,
            out_channels=1,
            loss_decode=dict(
                type="CrossEntropyLoss",
                use_sigmoid=True,
                loss_weight=0.4,
            ),
        ),
        dict(
            type="FCNHead",
            in_channels=32,
            channels=64,
            num_convs=2,
            num_classes=num_classes,
            in_index=2,
            norm_cfg=norm_cfg,
            concat_input=False,
            align_corners=False,
            out_channels=1,
            loss_decode=dict(
                type="CrossEntropyLoss",
                use_sigmoid=True,
                loss_weight=0.4,
            ),
        ),
        dict(
            type="FCNHead",
            in_channels=64,
            channels=256,
            num_convs=2,
            num_classes=num_classes,
            in_index=3,
            norm_cfg=norm_cfg,
            concat_input=False,
            align_corners=False,
            out_channels=1,
            loss_decode=dict(
                type="CrossEntropyLoss",
                use_sigmoid=True,
                loss_weight=0.4,
            ),
        ),
        dict(
            type="FCNHead",
            in_channels=128,
            channels=1024,
            num_convs=2,
            num_classes=num_classes,
            in_index=4,
            norm_cfg=norm_cfg,
            concat_input=False,
            align_corners=False,
            out_channels=1,
            loss_decode=dict(
                type="CrossEntropyLoss",
                use_sigmoid=True,
                loss_weight=0.4,
            ),
        ),
    ],
)

work_dir = "test_output"

train_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(type="LoadAnnotations"),
    dict(type="Resize", img_scale=(800, 800), ratio_range=(0.5, 2.0)),
    dict(type="RandomCrop", crop_size=(600, 600), cat_max_ratio=0.75),
    dict(type="RandomFlip", prob=0.5),
    dict(type="PhotoMetricDistortion"),
    dict(type="Normalize", **img_norm_cfg),
    # dict(type="Pad", size=crop_size, pad_val=0, seg_pad_val=255),
    dict(type="DefaultFormatBundle"),
    dict(type="Collect", keys=["img", "gt_semantic_seg"]),
]
test_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(
        type="MultiScaleFlipAug",
        img_scale=(800, 800),
        img_ratios=[0.5, 0.75, 1.0, 1.25, 1.5, 1.75],
        flip=False,
        transforms=[
            dict(type="Resize", keep_ratio=True),
            dict(type="RandomFlip"),
            dict(type="Normalize", **img_norm_cfg),
            dict(type="ImageToTensor", keys=["img"]),
            dict(type="Collect", keys=["img"]),
        ],
    ),
]

data = dict(
    samples_per_gpu=8,  # When BN is used, we must have >= 2 samples per GPU
    workers_per_gpu=1,  # Might be >= 1 ?
    train=dict(
        # type="MGIDataset",
        data_root="../dataset/",
        img_dir="images/train",
        ann_dir="masks/train",
        img_suffix=".jpg",
        seg_map_suffix=".png",
        pipeline=train_pipeline,  # If not overridden uses the one inherited from _base_
    ),
    val=dict(
        # type="MGIDataset",
        data_root="../dataset",
        img_dir="images/val",
        ann_dir="masks/val",
        img_suffix=".jpg",
        seg_map_suffix=".png",
        pipeline=test_pipeline,
    ),
    test=dict(
        # type="MGIDataset",
        data_root="../dataset",
        img_dir="images/test",
        ann_dir="masks/test",
        img_suffix=".jpg",
        seg_map_suffix=".png",
        pipeline=test_pipeline,
    ),
)
```

As well as custom dataset definition:

```python
@DATASETS.register_module()
class MGIDataset(CustomDataset):
    CLASSES = ("sane", "sick")
    PALETTE = [[0, 255, 0], [255, 0, 0]]

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        assert osp.exists(self.img_dir)
```

The `.png` files are 8-bit grayscale images with 0s and 1s. 
To confirm any doubts, this code:

```python
img = mmcv.imread("../dataset/images/test/62.jpg")
model.cfg = cfg
model.CLASSES = ("sane", "sick")
model.eval()
result = inference_segmentor(model, img)
print(result)

plt.figure(figsize=(8, 6))
show_result_pyplot(model, img, result, palette)
```

Produces this output:

```
[array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]], dtype=float32)]
```

Any help appreciated, thanks in advance

**Comment by coccocarmiano** [comment]
Update: I was using images with all pixel labeled as 0 and some containing spots of labels 1. This is to be done carefully because the model may decide that always predicting 0 is a good strategy. The fix for now has been to remove the "no anomaly" images, and it seems to work.
